deepchecks/recommender/checks/cold_start_detection.py

```diff
@@ -0,0 +1,101 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2023 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+from typing import List, Union
+import pandas as pd
+import numpy as np
+from PyNomaly import loop
+from deepchecks.core import CheckResult, ConditionCategory, ConditionResult
+from deepchecks.core.errors import (DeepchecksProcessError, DeepchecksTimeoutError, DeepchecksValueError,
+                                    NotEnoughSamplesError)
+from deepchecks.tabular import  SingleDatasetCheck
+from deepchecks.recommender import  Context, InteractionDataset
+from deepchecks.utils.dataframes import select_from_dataframe
+from deepchecks.utils.strings import  format_percent
+from deepchecks.utils.typing import Hashable
+import plotly.express as px
+
+__all__ = ['ColdStartDetection']
+
+
+
+class ColdStartDetection(SingleDatasetCheck):
+    """ Retrieve cold start users and items. \
+        Cold start entity refers to new entity who have limited or no historical data or interactions within a system or platform. \
+        They have not yet established a significant user/item history that can be leveraged for personalized recommendations or targeted actions. \
+        For the model, there can be some consequences fromincreaseduncertainty to the difficulty in personalization. \
```

Comment:
fix "fromincreaseduncertainty" and shorten this description a bit.

Reply:
https://github.com/deepchecks/deepchecks/commit/8953bac1afa2bd08e87eccbafaec02505be0c8a8